include/bl_storage.h

```diff
@@ -106,6 +158,151 @@ uint16_t get_monotonic_counter(void);
  */
 int set_monotonic_counter(uint16_t new_counter);
 
+/**
+ * @brief The PSA life cycle states a device can be in.
+ *
+ * The LCS can be only transitioned in the order they are defined here.
+ */
+enum lcs {
+	BL_STORAGE_LCS_UNKNOWN = 0,
+	BL_STORAGE_LCS_ASSEMBLY = 1,
+	BL_STORAGE_LCS_PROVISIONING = 2,
+	BL_STORAGE_LCS_SECURED = 3,
+	BL_STORAGE_LCS_DECOMMISSIONED = 4,
+};
+
+/**
+ * Copies @p src into @p dst. Reads from @p src are done 32 bits at a
+ * time. Writes to @p dst are done a byte at a time.
+ *
+ * This ensures UICR->OTP's word-aligned and word-sized read
+ * requirement is met and also allows the destination buffer to be
+ * free of this requirement.
+ *
+ * @param[out] dst destination buffer
+ * @param[in] src source buffer
+ * @param[in] size number of *bytes* in src to copy into dst
+ */
+NRFX_STATIC_INLINE void otp_copy32(uint8_t *restrict dst, uint32_t volatile *restrict src,
+				   size_t size)
+{
+	for (int i = 0; i < size / 4; i++) {
+		uint32_t val = src[i];
+
+		for (int j = 0; j < 4; j++) {
+			dst[i * 4 + j] = (val >> 8 * j) & 0xFF;
+		}
+	}
+}
+
+/* OTP is 0xFFFF after erase so setting all bits to 0 so a full erase is needed
+ * to reset
+ */
+#define STATE_ENTERED 0x0000
+#define STATE_NOT_ENTERED 0xFFFF
+
+/* The below bl_storage functions are static inline in the header file
```

Comment:
```suggestion
/* The bl_storage functions below are static inline in the header file
```
or The following ...